runtime/lua/vim/shared.lua

```diff
@@ -737,6 +737,51 @@ function vim.list_slice(list, start, finish)
   return new_list
 end
 
+--- Efficiently insert items into the middle of a list.
+---
+--- Calling table.insert() in a loop will re-index the tail of the table on
+--- every iteration, instead this function will re-index  the table exactly
+--- once.
+---
+--- Based on https://stackoverflow.com/questions/12394841/safely-remove-items-from-an-array-table-while-iterating/53038524#53038524
+---
+---@param t any[]
+---@param first integer
+---@param last integer
+---@param v any
+function vim.list_insert(t, first, last, v)
```

Comment:
Can you not expose this by prefixing it with an `_`.

Reply:
Do you mean to consider it as an internal API? Then I will also remove it from the commit messages.

Reply:
Yes please.